### ISSUE TITLE
Error deep cloning with named association

### ISSUE THREAD

**Issue by gangleton** [moiristo/deep_cloneable#67]
I'm trying to deep clone an object with several associations:

```
Poll has_many :questions
Question has_many :options
Option  has_many :follow_up, :class_name => "Question", :foreign_key => "parent_option_id", :dependent => :destroy, :inverse_of => :option
```

When I try to deep clone a poll object, `poll.deep_clone include:
                      { questions:
                        [
                          {options: :follow_up}
                        ]
                       }
`, I get the error `NoMethodError: undefined method 'each' for #<Option:0x007fd87d419490>`, 

with the following trace:

```
activemodel-3.2.22.2/lib/active_model/attribute_methods.rb:407:in `method_missing'
activerecord-3.2.22.2/lib/active_record/attribute_methods.rb:149:in `method_missing'
activerecord-3.2.22.2/lib/active_record/associations/collection_association.rb:310:in `replace'
activerecord-3.2.22.2/lib/active_record/associations/collection_association.rb:41:in `writer'
activerecord-3.2.22.2/lib/active_record/associations/builder/association.rb:51:in `block in define_writers'
deep_cloneable-2.2.1/lib/deep_cloneable.rb:127:in `block in dup_has_many_association'
activerecord-3.2.22.2/lib/active_record/associations/collection_proxy.rb:91:in `collect'
activerecord-3.2.22.2/lib/active_record/associations/collection_proxy.rb:91:in `method_missing'
deep_cloneable-2.2.1/lib/deep_cloneable.rb:124:in `dup_has_many_association'
```

It seems to be an issue when the association doesn't use the same name as the class of the object, similar to https://github.com/moiristo/deep_cloneable/issues/58

Any idea how to get around this?

**Comment by moiristo** [comment]
You're `follow_up` association looks a bit mangled, what's going on there?

* `follow_up` is `has_many`, therefore it should be `follow_ups`, or it should be a `has_one` association instead
* A follow_up is a `Question`? Then why is its foreign key id called `parent_option_id`?

What actually happens (I think) is the following:

* An `Option` record is found and cloned
* The `follow_up` association is found and cloned
* In the cloned `FollowUp`, deep_cloneable passes the  cloned option to the inverse association (should be something like `belongs_to :option, inverse_of: :follow_up` in follow_up)
* For some reason, AR expects a collection of objects instead of a single `Option` object. That's what the error shows. Therefore I think your associations are not correctly specified.

**Comment by gangleton** [comment]
`follow_up` is a naming problem that I inherited with this code base, I know it's not correct. 

A `follow_up` is a question that belongs to an option. Options have two associations with Questions- one as a `belongs_to`, and one as `has_many`. The other side of that is 
```
Question belongs_to :option, :foreign_key => "parent_option_id", inverse_of: :follow_up
```
The foreign key is stored on Question.

**Comment by moiristo** [comment]
Ok, I need to reproduce the error in the test suite first in order to fix it. I'll try to add one shortly.. unless you'd like to give it a try yourself of course ;)

**Comment by gangleton** [comment]
I  should be able to do that this afternoon…

**Comment by moiristo** [comment]
This should be fixed by #71 (release 2.2.2).
